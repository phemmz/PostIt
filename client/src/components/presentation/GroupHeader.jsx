import React from 'react';
import { Link } from 'react-router';

/**
 * Home
 * @return {*} div
 */
const GroupHeader = ({ notifications }) => {
  return (
    <div className="row">
      <div className="col s12">
        <h5 className="white-text text-darken-4">Your Group List
          <Link to="/group">
            <i
              className="fa fa-plus-square add-group tooltipped"
              data-tooltip="Create New Group"
              aria-hidden="true"
            />
          </Link>
          { (notifications.length !== 0) ?
            (
              <Link to="/notification" className="mybell on-notification">
                <i
                  className="fa fa-bell-o tooltipped"
                  data-tooltip="You have new Notification(s)"
                  aria-hidden="true"
                />
              </Link>
            ) :
            (
              <Link to="/notification" className="mybell">
                <i
                  className="fa fa-bell-o tooltipped"
                  data-tooltip="No new Notification"
                  aria-hidden="true"
                />
              </Link>
            )
          }
        </h5>
      </div>
    </div>
  );
};

export default GroupHeader;
