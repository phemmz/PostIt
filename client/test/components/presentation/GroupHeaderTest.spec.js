import React from 'react';
import { mount } from 'enzyme';
import GroupHeader from '../../../src/components/presentation/GroupHeader';

describe('Message Component', () => {
  const notifications = ['New message from neene'];
  const noNotification = [];
  const children = (
    <div className="scrbar center">
      <h5>You currently dont belong to any group</h5>
    </div>);
  it('displays the group header with notifications in the group section', () => {
    const wrapper = mount(
    <GroupHeader
      notifications={notifications}
    >{children}</GroupHeader>);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.prop('notifications')).toBe(notifications);
    expect(wrapper.find('div').exists()).toBe(true);
    expect(wrapper.find('h5').exists()).toBe(true);
    expect(wrapper.find('i').exists()).toBe(true);
    expect(wrapper.find('hr').exists()).toBe(true);
    expect(wrapper.find('div').length).toBe(7);
    expect(wrapper.find('hr').length).toBe(1);
    expect(wrapper.find('h5').length).toBe(2);
    expect(wrapper.find('i').length).toBe(2);
  });
  it('displays the group header without notifications', () => {
    const wrapper = mount(
    <GroupHeader
      notifications={noNotification}
    >{children}</GroupHeader>);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.prop('notifications')).toBe(noNotification);
    expect(wrapper.find('div').exists()).toBe(true);
    expect(wrapper.find('h5').exists()).toBe(true);
    expect(wrapper.find('i').exists()).toBe(true);
    expect(wrapper.find('hr').exists()).toBe(true);
    expect(wrapper.find('div').length).toBe(7);
    expect(wrapper.find('hr').length).toBe(1);
    expect(wrapper.find('h5').length).toBe(2);
    expect(wrapper.find('i').length).toBe(2);
  });
});
