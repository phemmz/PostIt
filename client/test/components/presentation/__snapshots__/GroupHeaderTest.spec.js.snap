// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Component displays the group header with notifications in the group section 1`] = `
<GroupHeader
  notifications={
    Array [
      "New message from neene",
    ]
  }
>
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <div
        className="col s12 m12"
      >
        <div
          className="container-fluid"
        >
          <div
            className="row"
          >
            <div
              className="col s12"
            >
              <h5
                className="white-text text-darken-4"
              >
                Your Group List
                <Link
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/group"
                >
                  <a
                    onClick={[Function]}
                    style={Object {}}
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-plus-square add-group tooltipped"
                      data-tooltip="Create New Group"
                    />
                  </a>
                </Link>
                <Link
                  className="mybell on-notification"
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/notification"
                >
                  <a
                    className="mybell on-notification"
                    onClick={[Function]}
                    style={Object {}}
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-bell-o tooltipped"
                      data-tooltip="You have new Notification(s)"
                    />
                  </a>
                </Link>
              </h5>
            </div>
          </div>
          <hr />
          <div
            className="scrbar center"
          >
            <h5>
              You currently dont belong to any group
            </h5>
          </div>
        </div>
      </div>
    </div>
  </div>
</GroupHeader>
`;

exports[`Message Component displays the group header without notifications 1`] = `
<GroupHeader
  notifications={Array []}
>
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <div
        className="col s12 m12"
      >
        <div
          className="container-fluid"
        >
          <div
            className="row"
          >
            <div
              className="col s12"
            >
              <h5
                className="white-text text-darken-4"
              >
                Your Group List
                <Link
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/group"
                >
                  <a
                    onClick={[Function]}
                    style={Object {}}
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-plus-square add-group tooltipped"
                      data-tooltip="Create New Group"
                    />
                  </a>
                </Link>
                <Link
                  className="mybell"
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/notification"
                >
                  <a
                    className="mybell"
                    onClick={[Function]}
                    style={Object {}}
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-bell-o tooltipped"
                      data-tooltip="No new Notification"
                    />
                  </a>
                </Link>
              </h5>
            </div>
          </div>
          <hr />
          <div
            className="scrbar center"
          >
            <h5>
              You currently dont belong to any group
            </h5>
          </div>
        </div>
      </div>
    </div>
  </div>
</GroupHeader>
`;
